anndata/_core/sparse_dataset.py

```diff
@@ -474,6 +474,15 @@ def append(self, sparse_matrix: ss.spmatrix):
                 f"Matrices must have same format. Currently are "
                 f"{self.format!r} and {get_format(sparse_matrix)!r}"
             )
+        indptr_offset = self.group["indptr"][-1]
+        if indptr_offset.dtype == np.int32:
```

Comment:
Why can't we check the dtype of the group directly?  This will read in from `indptr` every time which doesn't necessarily have to happen if the `assert` checks below don't pass.  Is there a reason not to do this via the cached property?